services/ad-tech/src/controllers/event-report-store.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ Copyright 2022 Google LLC
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+/** Basic categories of event-level reports. */
+export enum EventReportCategory {
+  EVENT_LEVEL_LOG,
+  ARA_EVENT_LEVEL,
+  ARA_EVENT_LEVEL_DEBUG,
+  ARA_AGGREGATE,
+  ARA_AGGREGATE_DEBUG,
+  PAGG_VIA_SS,
+  PAGG_VIA_SS_DEBUG,
+  PAGG_VIA_PAAPI,
+  PAGG_VIA_PAAPI_DEBUG,
+}
+
+/** High-level abstraction of an event-level report. */
+export interface EventReport {
+  category: EventReportCategory;
+  timestamp: string;
+  data: any;
+}
+
+export const EventReportStore = (() => {
+  // In-memory storage for reports.
+  const Reports: EventReport[] = [];
+  // Clear in-memory storage every 10 min
+  setInterval(() => {
```

Comment:
This module is not specific to ARA. This module is meant to be an in-memory store for all kinds of reports -- logs from reportResult, reportWin, beacon reports, ARA event-level, and some aggregate reports too. And since it's just an in-memory store, we clear the store every 10 mins. Ideally, this should be a cache with fixed TTL per item.